File: Projects/codechef/CHPINTU.java
package codechef;

import java.util.Scanner;

class CHPINTU {
    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int t = stdin.nextInt();
        while (t-- > 0) {
            int n = stdin.nextInt();
            int m = stdin.nextInt();
            int[] fs = new int[n];
            boolean[] isAvailables = new boolean[m];
            for (int i = 0; i < n; i++) {
                int f = stdin.nextInt();
                fs[i] = f;
                isAvailables[f - 1] = true;
            }
            int[] costs = new int[m];
            for (int i = 0; i < n; i++) {
                costs[fs[i] - 1] += stdin.nextInt();
            }
            int minCost = Integer.MAX_VALUE;
            for (int i = 0; i < m; i++) {
                if (!isAvailables[i]) continue;
                minCost = Math.min(minCost, costs[i]);
            }
            System.out.println(minCost);
        }
        stdin.close();
    }
}

File: Projects/codechef/CHPINTU.kt
package codechef

import kotlin.Throws
import codechef.KNPSK.Record

internal object CHPINTU {
    fun main(args: Array<String?>?) {
        val stdin = Scanner(System.`in`)
        var t: Int = stdin.nextInt()
        while (t-- > 0) {
            val n: Int = stdin.nextInt()
            val m: Int = stdin.nextInt()
            val fs = IntArray(n)
            val isAvailables = BooleanArray(m)
            for (i in 0 until n) {
                val f: Int = stdin.nextInt()
                fs[i] = f
                isAvailables[f - 1] = true
            }
            val costs = IntArray(m)
            for (i in 0 until n) {
                costs[fs[i] - 1] += stdin.nextInt()
            }
            var minCost: Int = Integer.MAX_VALUE
            for (i in 0 until m) {
                if (!isAvailables[i]) continue
                minCost = Math.min(minCost, costs[i])
            }
            System.out.println(minCost)
        }
        stdin.close()
    }
}